launchers/cache_serve.go

```diff
@@ -51,6 +54,21 @@ func CacheServe(ctx context.Context, engine *gin.Engine, egrp *errgroup.Group, m
 		return nil, err
 	}
 
+	// Check for the sentinel file
+	if param.Cache_SentinelLocation.IsSet() {
+		sentinelPath := param.Cache_SentinelLocation.GetString()
```

Comment:
This seems redundant. Can the server_utils function be made more general to check both the cache and origin sentinel locations?

Reply:
The first draft (not published) was to take advantage of the `server_utils` function and make them share the same core logic, but I figured the real "core" logic is just a `os.Stat` call. Given that the error message has different format for cache and origin, and that origin and cache reads from different parameters, I figured I'd be ok to have a separate function to handle them